lib/connection.ex

```diff
@@ -254,40 +205,15 @@ defmodule Janus.Connection do
            "transaction" => transaction,
            "error" => %{"code" => code, "reason" => reason}
          },
-         state(pending_calls_table: pending_calls_table) = s
+         state(pending_calls_table: pending_calls_table) = state
        ) do
-    case :ets.lookup(pending_calls_table, transaction) do
-      [{_transaction, from, expires_at}] ->
-        if DateTime.utc_now() |> DateTime.to_unix(:millisecond) > expires_at do
-          Logger.warn(
-            "[#{__MODULE__} #{inspect(self())}] Received error reply to the outdated call: transaction = #{
-              inspect(transaction)
-            }, code = #{inspect(code)}, reason = #{inspect(reason)}"
-          )
-
-          :ets.delete(pending_calls_table, transaction)
-          {:ok, s}
-        else
-          Logger.warn(
-            "[#{__MODULE__} #{inspect(self())}] Call error: transaction = #{inspect(transaction)}, code = #{
-              inspect(code)
-            }, reason = #{inspect(reason)}"
-          )
-
-          GenServer.reply(from, {:error, {:gateway, code, reason}})
-          :ets.delete(pending_calls_table, transaction)
-          {:ok, s}
-        end
-
-      [] ->
-        Logger.warn(
-          "[#{__MODULE__} #{inspect(self())}] Received error reply to the unknown call: transaction = #{
-            inspect(transaction)
-          }, code = #{inspect(code)}, reason = #{inspect(reason)}"
-        )
+    Transaction.handle_transaction(
```

Comment:
Dialyzer complains here

Reply:
Also, you're ignoring the result of this function, including errors it may return

Reply:
All errors are already handled in the transaction module.
 - unknown transaction - logged as warn, I don't think any other action is needed here
 - outdated transaction - logged as warn, will result in timeout error in the Connection.call module by design of that function

Reply:
I don't think any error is ignored here really. All that we expect are handled level deeper. Also none of those errors will affect connection process or connection state.